### PR TITLE
Binary search in rust

### LINKED ISSUE #2175: Binary Search
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart - @keshakaneria
- [x] Go
- [x] Java
- [x] Julia
- [x] JS
- [x] Kotlin
- [x] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [x] Rust - @raksha009
- [x] TypeScript - @Shr03mink

### PR DESCRIPTION
**Pull request by raksha009**
Fixes #2175 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- added new code
- 

#### Languages Used:
- Rust
- 

#### Files Added:
- Binary_search.rs
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raksha009** [comment]
plz review my pr @singh-shreya6  maam

**Comment by raksha009** [comment]
@MastersAbh I have followed 4 space indentation only. Plz give an example where my file is not indented properly.

File: Binary_Search/Binary_Search.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    let mut leftpart = 0;
+    let mut rightpart = size - 1;
+    let mut midindex = 0;
+    let mut checkpoint = 0;
+    while leftpart <= rightpart {
+    	midindex = leftpart + (rightpart - leftpart) / 2;
+
+    	// Element found
+    	if vector[midindex] == element{
+    		checkpoint = 1;
```
Comment (singh-shreya6) [inline Binary_Search/Binary_Search.rs:38]: Indentation is more than 4 spaces here.

File: Binary_Search/Binary_Search.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    		println!("Element is found at index ");
+    		println!("{:?}" , midindex + 1);
+    		break;
+    	}
+
+    	else if vector[midindex] < element{
+    		leftpart = midindex + 1;
```
Comment (singh-shreya6) [inline Binary_Search/Binary_Search.rs:45]: Same here.

File: Binary_Search/Binary_Search.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    	}
+
+    	else {
+    		rightpart = midindex - 1;
```
Comment (singh-shreya6) [inline Binary_Search/Binary_Search.rs:49]: More than 4 spaces.

File: Binary_Search/Binary_Search.rs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    	}
+         
+    }
+    
+    if  checkpoint == 0 {
+    	println!("Element is not found in the array ");
```
Comment (singh-shreya6) [inline Binary_Search/Binary_Search.rs:55]: Less than 4 spaces.

**Comment by raksha009** [comment]
done the changes @singh-shreya6 @MastersAbh

**Comment by singh-shreya6** [comment]
@MastersAbh Please review.

**Comment by raksha009** [comment]
Plz merge my PR @singh-shreya6 maam

**Pull request merged**